ts/components/SelectLogoutOption.tsx

```diff
@@ -68,42 +68,40 @@ class SelectLogoutOption extends React.PureComponent<Props, State> {
   };
 
   public render() {
-    // Using the loading spinner HOC to avoid reimplementing the Loading component
-    // TODO - this part can be improved within this story https://www.pivotaltracker.com/story/show/169185930
-    const ContainerComponent = withLoadingSpinner(() => (
-      <Container>
-        <Content style={styles.content}>
-          {this.props.header || null}
-          <List>
-            <ListItemComponent
-              title={I18n.t("profile.logout.cta.keepData.title")}
-              subTitle={I18n.t("profile.logout.cta.keepData.description")}
-              onPress={() => this.logout({ keepUserData: true })}
-              useExtendedSubTitle={true}
-            />
+    return (
+      <LoadingSpinnerOverlay isLoading={this.state.isLoading}>
```

Comment:
I suggest here to come back to the previous approach: with the applied changes (on this file) you stop using the HOC component and you need to explicitly wrap the content of the screen (`<Container>...</Container>`) into the `<LoadingSpinnerOverlay>`, while it is mean to be the scope of `withLoadingSpinner`